great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+import abc
+import enum
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from great_expectations.exceptions import ProfilerConfigurationError
+
+
+@dataclass
+class CardinalityLimit(abc.ABC):
+    name: str
+
+
+@dataclass
+class RelativeCardinalityLimit(CardinalityLimit):
+    max_proportion_unique: Union[int, float]
+    related_metric_name: str = "column.unique_proportion"
+
+
+@dataclass
+class AbsoluteCardinalityLimit(CardinalityLimit):
+    max_unique_values: int
+    related_metric_name: str = "column.distinct_values.count"
```

Comment:
@anthonyburdi `related_metric_name` or `relevant_metric_name`? :)

Reply:
I don't like either upon reflection. Changing to `metric_name_defining_limit` - thoughts?